firestore/google/cloud/firestore_v1/_helpers.py

```diff
@@ -760,14 +760,8 @@ def apply_merge(self, merge):
 
     def _get_update_mask(self, allow_empty_mask=False):
         # Mask uses dotted / quoted paths.
-        mask_paths = [
-            field_path.to_api_repr()
-            for field_path in self.merge
-            if field_path not in self.transform_merge
-        ]
-
-        if mask_paths or allow_empty_mask:
-            return common_pb2.DocumentMask(field_paths=mask_paths)
+        mask_paths = [field_path.to_api_repr() for field_path in self.merge]
+        return common_pb2.DocumentMask(field_paths=mask_paths)
```

Comment:
I can see your logic here. We just need to ensure the proper test coverage.

Reply:
yes, you can find a respective test case in test__helpers.py file and coverage is also 100%